src/Microsoft.OData.Core/RawValueWriter.cs

```diff
@@ -154,6 +154,87 @@ internal void Flush()
             }
         }
 
+        /// <summary>
+        /// Asynchronously start writing a raw output. This should only be called once.
+        /// </summary>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        internal async Task StartAsync()
+        {
+            if (this.settings.HasJsonPaddingFunction())
+            {
+                await this.textWriter.WriteAsync(this.settings.JsonPCallback)
+                    .ConfigureAwait(false);
+                await this.textWriter.WriteAsync(JsonConstants.StartPaddingFunctionScope)
+                    .ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Asynchronously end the writing of a raw output. This should be the last thing called.
+        /// </summary>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        internal Task EndAsync()
+        {
+            if (this.settings.HasJsonPaddingFunction())
+            {
+                return this.textWriter.WriteAsync(JsonConstants.EndPaddingFunctionScope);

```

Comment:
General question for all the Task methods in the PR; 
Do we have a general rule on where to elide and where not to elide async awaits with regards to Tasks?

Reply:
When a method is a pass-through. The reasoning here is that `HasJsonPaddingFunction` would never be expected to throw an exception. Here is how it is defined:
```csharp
        internal bool HasJsonPaddingFunction()
        {
            return !string.IsNullOrEmpty(this.JsonPCallback);
        }
```
By that reasoning, `EndAsync` method can be treated as a pass-through since we have the option of either returning the task that does the actual writing of the Json padding function closing parens (`)`) or a completed task - `TaskUtils.CompletedTask` is defined as a `static`. Either of the option saves us from creating a `Task` object hence this is an optimization.
It's unwise to elide where a function has multiple statements where any single one of them can throw. In this scenario, the function needs to the `async` modifier to cause the generation of the state machine class such that in the event of an exception being thrown, it'll be stashed in the task using the state machine's `SetException` method.